gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpGremlinEndpointHandler.java

```diff
@@ -147,40 +136,43 @@ public void channelRead(final ChannelHandlerContext ctx, final Object msg) {
                 return;
             }
 
-            final Quartet<String, Map<String, Object>, String, Map<String, String>> requestArguments;
+            final RequestMessage requestMessage;
             try {
-                requestArguments = HttpHandlerUtil.getRequestArguments(req);
+                requestMessage = HttpHandlerUtil.getRequestMessageFromHttpRequest(req);
             } catch (IllegalArgumentException iae) {
                 HttpHandlerUtil.sendError(ctx, BAD_REQUEST, iae.getMessage(), keepAlive);
                 ReferenceCountUtil.release(msg);
                 return;
             }
 
-            final String acceptString = Optional.ofNullable(req.headers().get("Accept")).orElse("application/json");
-            final Pair<String, MessageTextSerializer<?>> serializer = chooseSerializer(acceptString);
+            final String acceptMime = Optional.ofNullable(req.headers().get(HttpHeaderNames.ACCEPT)).orElse("application/json");
+            final Pair<String, MessageTextSerializer<?>> serializer = chooseSerializer(acceptMime);
```

Comment:
This is not exactly related to PR. However, shouldn't we need to drop the assumption that HTTP server can support only `MessageTextSerializer`?

Reply:
yes, i think my previous comment addresses that. `MessageTextSerializer` really isn't a necessary anymore.